src/legacy/core_plugins/kibana/public/discover/application.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import angular from 'angular';
+
+/**
+ * Here's where Discover's inner angular is mounted and rendered
+ */
+export async function renderApp(moduleName: string, element: HTMLElement) {
+  require('./angular');
+  const $injector = mountDiscoverApp(moduleName, element);
+  return () => $injector.get('$rootScope').$destroy();
+}
+
+function mountDiscoverApp(moduleName: string, element: HTMLElement) {
+  const mountpoint = document.createElement('span');
+  // eslint-disable-next-line
+  mountpoint.innerHTML = `<span ng-view></span>`;
```

Comment:
Maybe slightly longer code, but it would help us not needing to use `innerHTML` here, if we would also create that element with document APIs and setAttribute on it and than attach it also the one above?

If we want to keep it with the innerHTML variant, I would prefer if we could write a short explanation why it;s safe (because we’re setting a static simple string).

Reply:
that's a good idea, I've changed that!